qiskit/transpiler/passes/basis/unroller.py

```diff
@@ -70,6 +70,9 @@ def run(self, dag):
 
             # TODO: allow choosing other possible decompositions
             try:
+                phase = 0
```

Comment:
Could you drop `phase` and just accumulate into `dag.global_phase` directly? There doesn't seem to be a case where it doesn't get added in.

Reply:
i think there is a case when we end up in lines [101-103](https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/transpiler/passes/basis/unroller.py#L101-L103). 
if we accumulate into `dag.global_phase` directly we will add twice the `node.op.definition.global_phase`.